lib/awful/widget/calendar_popup.lua

```diff
@@ -36,7 +36,7 @@ local beautiful = require("beautiful")
 
 local calendar_popup = { offset = 0, mt = {} }
 
-local properties = { "markup", "fg_color", "bg_color", "shape", "padding", "border_width", "border_color", "opacity" }
+local properties = { "fg_color", "bg_color", "markup", "shape", "padding", "border_width", "border_color", "opacity" }
```

Comment:
`properties` are only used in [for-ipairs](https://github.com/awesomeWM/awesome/blob/03a815dfce0cd8b79ae9152fcf22e5af46ba5a02/lib/awful/widget/calendar_popup.lua#L119), so if the table order is changed, it is no problem in other codes. When `prop` is "markup" and `cell` is "focus", `default` will use `string.format`. The `string.format` depends on `props.fg_color` and `props.fg_color`. Setting values to `props.fg_color`, `props.fg_color`, and `props.markup` in this order prevents `nil` from becoming an argument of `string.format`.

If someone changes the order of `properties`, the action may change. If you are afraid of the problem, I write like this after for-ipairs.
```lua
if cell == "focus" and props.markup == nil then
    props.markup = string.format(
        '<span foreground="%s" background="%s"><b>%s</b></span>',
        props.fg_color, props.bg_color, "%s"
    )
end
```
Which way is better?

Reply:
Oh, ok. I wasn't aware this can be called with nil props. In this case, I would indeed recommend testing properties before using them. I don't know whether testing the nullity of `props.markup` would help in this situation, tho.

I think you can split the final string to format it with existing values only :

```lua
local fg = props.foreground and string.format('foreground="%s"', props.foreground) or ""
local bg = -- ...
props.markup = string.format(
    '<span %s %s><b>%s</b></span>',
    fg, bg, "%s"
)
```

Reply:
Code in your review looks good. So I used it. 

I thought again. As a result, it looked better not to change the order of `properties`. So I wrote that way. 

I have tested whether `props.markup` is nil or not, because I want to prefer the user-defined function and format string.

Reply:
> I have tested whether props.markup is nil or not, because I want to prefer the user-defined function and format string.

Yes, you're right.